hooks/src/index.js

```diff
@@ -6,6 +6,9 @@ let currentIndex;
 /** @type {import('./internal').Component} */
 let currentComponent;
 
+/** @type {number} */
+let currentHook = 0;
```

Comment:
I'm wondering about the index association here: would it be feasible to instead set+pass a reference to the current hook function?

Preact/debug could then `import * from hooks` and do a reverse lookup to get the hook name:

```js
import { * as hooks } from 'preact/hooks';

options._hook = (comp, index, type) = {
  let hookName;
  for (let i in hooks) if (hooks[i]===type) {
    hookName = i;
    break;
  }
  // Etc
};
```

Reply:
In the current architecture the whole options handling is injected by the devtools. Previously we did this in `preact/debug`, but it made us less flexible regarding bug fixes. Whenever there is a bug in the devtools bridge we had to tell users to upgrade their Preact version which feels wrong to me.

Because it's all handled by the extension in a central place we can easily roll out fixes for all supported Preact versions.

(For everyone reading this: We discussed various alternatives in Slack but came to the conclusion that the approach in this PR is the best one so far. We'll move forward with this)